BUILD

```diff
@@ -2231,6 +2231,28 @@ grpc_cc_library(
     ],
 )
 
+grpc_cc_library(
+    name = "grpc_rpc_encoding",
+    srcs = [
+        "src/cpp/ext/filters/census/rpc_encoding.cc",
+    ],
+    hdrs = [
+        "src/cpp/ext/filters/census/rpc_encoding.h",
+    ],
+    external_deps = [
+        "absl/base",
+        "absl/base:core_headers",
+        "absl/meta:type_traits",
+        "absl/status",
+        "absl/strings",
+        "absl/time",
+    ],
+    language = "c++",
+    tags = ["nofixdeps"],
+    visibility = ["//visibility:public"],
```

Comment:
`@grpc:public` is still too broad. This needs to be for _internal_ use only. I didn't realize before that there's an issue here that this visibility needs to work across both google-internal and github. This will likely require making some changes to the `grpc_cc_library` macro definition.

You'll need to add something like `@grpc:grpc_python_observability` which is included [here](https://github.com/grpc/grpc/blob/7b74b078854e8e683f5a441ab7e41c5c6388e1c0/bazel/grpc_build_system.bzl#L87) as well as the internal version of this file.

The reason here is that if we leave this public then it is a signal to all Bazel users that this is a _public API_ and we would be on the hook to maintain it as such going forward, including writing gRFCs for all additions to it.

Reply:
ACK, will change it to `@grpc:grpc_python_observability`